_docs/cardano/2017-01-04-transactions.md

```diff
@@ -3,41 +3,113 @@ layout: default
+transaction `Tx1`, the node under _Alice_' wallet `N1`, and the node under _Bob_'s wallet `N2`.
+
+So, the node `N1` do these steps:
+
+1. [creates](https://github.com/input-output-hk/cardano-sl/blob/63adb31e813e21ec9da21cfa69984840308bbfa2/src/Pos/Wallet/Tx.hs#L41) transaction `Tx1` and [signs](https://github.com/input-output-hk/cardano-sl/blob/63adb31e813e21ec9da21cfa69984840308bbfa2/src/Pos/Wallet/Tx/Pure.hs#L83) it with its private key,
+2. [sends](https://github.com/input-output-hk/cardano-sl/blob/63adb31e813e21ec9da21cfa69984840308bbfa2/src/Pos/Wallet/Tx.hs#L53) it to all known nodes (i.e. neighbors),
+3. [save](https://github.com/input-output-hk/cardano-sl/blob/63adb31e813e21ec9da21cfa69984840308bbfa2/src/Pos/Wallet/Tx.hs#L44) it in its local data.
+
+Each of `N1`'s neighbors sends transaction `Tx1` to its neighbors and so on, and some slot leader will store this transaction to the next block in the ledger.
+
+## Design
+
+Each transaction contains a list of _inputs_ and a list of _outputs_, and outputs of the transaction `Tx0` can be used as an inputs of the other transaction `Tx1` and so on:
+
+~~~
+            Tx0                           Tx1
+  +----------------------+      +----------------------+
+  |                      |      |                      |
+  |  Inputs     Outputs  |      |  Inputs     Outputs  |
+  | +------+   +-------+ |      | +------+   +-------+ |
+  | | In0  |   | Out0  + |      | | In0  |   | Out0  | |
+  | +------+   +-------+ |      | +------+   +-------+ |
+  | | In1  |   | Out1  | |      | | In1  |   | Out1  | |
+  | +------+   +-------+ |      | +------+   +-------+ |
+  | | ...  |   | ...   | |      | | ..   |   | ...   | |
+  | +------+   +-------+ |      | +------+   +-------+ |
+  | | InN  |   | OutN  | |      | | InN  |   | OutN  | |
+  | +------+   +-------+ |      | +------+   +-------+ |
+  |                      |      |                      |
+  +----------------------+      +----------------------+     ...
+~~~
+
+Inputs and outputs inform us about _money flow_: input informs where the money came from, and output informs where the money come to.
+
+So, each input contains:
+
+1. An id of transaction `TxN` which output is used for this input. Transaction's id is a BLAKE2s-256 hash of the transaction, something like `f9bcbe752aee4512457f1fd350200cf870906b7e6e836688c9a3779645c86c01`.
+2. An index of the using output in `TxN`'s outputs. This index is 32-bit unsigned integer.
+
+And each output contains:
 
-Transactions are signed with the issuer's secret key. All of this,
-including selection of unspent outputs to be used as inputs, is done
-automatically, and you don't have to worry about it when making a
-transaction using Daedalus UI.
+1. An address of the node `N` we want to send a value to. An address is a BLAKE2s-224 hash of the public key of the `N` node, something like `88ca7f79d4edcf911b60eeb96c8e9284d0c07c6e61c59a9e1c17a5e9`.
```

Comment:
Address is hash of hash of public key. It's described here: https://github.com/input-output-hk/cardano-docs.iohk.io/blob/csld23-transaction-semantic/_docs/cardano/2017-01-02-addresses.md#hashing